unstructured/documents/elements.py

```diff
@@ -505,6 +506,35 @@ def field_consolidation_strategies(cls) -> dict[str, ConsolidationStrategy]:
 _P = ParamSpec("_P")
 
 
+def assign_hash_ids(elements: List[Element]) -> List[Element]:
+    """Updates the `id` and `parent_id` attributes of each element
+    in the list of elements based on the element's attributes and its index in sequence
+
+    Args:
+        elements: The list of elements whose IDs are to be updated.
+
+    Returns:
+        The list of elements with updated IDs.
+    """
+
+    old_to_new_id_mapping = {}
+    for idx_in_seq, element in enumerate(elements):
+        old_id = element.id
+        new_id = element.id_to_hash(idx_in_seq)
+        old_to_new_id_mapping[old_id] = new_id
+
+    for element in elements:
```

Comment:
There are cases when elements are in the right order, that's why we need 2 passes.
![image](https://github.com/Unstructured-IO/unstructured/assets/64484917/8eb3d92b-3900-4eed-9a4b-29f45b3dfd81)

Reply:
This doesn't sound quite right to me. Let's discuss.

Two passes sounds right:
1. UUID -> Hash for all elements, saving mapping.
2. Update all parent-ids based on mapping.

I don't see why "already_updated_parent_ids()" is necessary. Wouldn't this get it done?

```python
old_to_new_mapping = {
    e.id: e.id_to_hash(idx)
    for idx, e in enumerate(elements)
}
for e in elements:
    parent_id = e.metadata.parent_id
    if not parent_id:
        continue
    e.metadata.parent_id = old_to_new_mapping(parent_id)
```